distribution/docs/src/main/resources/_contents/_developing/developing-intro-contents.adoc

```diff
@@ -0,0 +1,816 @@
+
+===== Load the Code Formatter Into the Eclipse IDE
+
+. Download the https://github.com/codice/ddf-support/blob/master/support-checkstyle/src/main/resources/ddf-eclipse-code-formatter.xml[Eclipse Code Formatter].
+. In Eclipse, select *Window → Preferences*. The Preferences window opens.
+. Select *Java → Code Style → Formatter*.
+. Select the *Edit...* button and then Select the downloaded file.
+. Select the *OK* button.
+
+===== Load the Code Formatter Into IntelliJ IDEA
+
+IntelliJ IDEA 13 is capable of importing Eclipse's Code Formatter directly from within IntelliJ without the use of any plugins.
+
+. Download the https://github.com/codice/ddf-support/blob/master/support-checkstyle/src/main/resources/ddf-intellij-code-formatter.jar[IntelliJ Code Formatter].
+. Open *IntelliJ* IDEA.
+. Select *File → Settings → Code Style → Java*.
+. Select *Manage*.
+. Select the *Import* button and select the file.
+
+===== Format Your Source Code Using Eclipse
+
+A developer may write code and format it before saving.
+
+. Before the file is saved, highlight all of the source code in the IDE editor window.
+. Right-click on the highlighted code.
+. Select *Source → Format*. The code formatter is applied to the source code and the file can be saved.
+
+====== Set Up Save Actions in Eclipse
+
+A developer can also set up Save Actions to format the source code automatically.
+
+. Open Eclipse.
+. Select *Window → Preferences* (*Eclipse → Preferences* on Mac). The Preferences window opens.
+. Select *Java → Editor → Save Actions*.
+. Select *Perform the selected actions on save*.
+. Select *Format source code*.
+. Select *Format all lines* or *Format edited lines*, as necessary.
+. Optionally, select *Organize imports* (recommended).
+. Select the *Apply* button.
+. Select the *OK* button.
+
+===== Format Source Code Using IntelliJ
+
+In the toolbar, select *Code → Reformat Code* or use the keyboard shortcut `Ctrl-Alt-L`.
+
+==== ${branding} Software Versioning
+
+${branding} follows the https://www.osgi.org/wp-content/uploads/SemanticVersioning.pdf[Semantic Versioning White Paper] for bundle versioning.
+
+==== Ensuring Compatibility
+
+===== Compatibility Goals
+
+The ${branding} framework, like all software, will mature over time.
+Changes will be made to improve efficiency, add features, and fix bugs.
+To ensure that components built for ${branding} and its sub-frameworks are compatible, developers must use caution when establishing dependencies from developed components.
+
+===== Guidelines for Maintaining Compatibility
+
+====== ${branding} Framework
+
+For components written at the ${branding} Framework level (see Developing at the Framework Level), adhere to the following specifications:
+
+[cols="2,1,5" options="header"]
+|===
+
+|Standard/Specification
+|Version
+|Current Implementation (subject to change)
+
+|OSGi Framework
+|4.2
+|Apache Karaf 2.x +
```

Comment:
We're currently on Karaf 4.0.5 (4.0.x). There are very significant differences between the 2.x and 4.0.x versions.